test/unit/setting_manager_test.rb

```diff
@@ -0,0 +1,56 @@
+require 'test_helper'
+
+class SettingManagerTest < ActiveSupport::TestCase
+  let(:setting_memo) { {} }
+  let(:category_memo) { { 'general' => N_('General') } }
+
+  setup do
+    Foreman::SettingManager.stubs(settings: setting_memo, categories: category_memo)
+  end
+
+  it 'adds setting to category general by default' do
+    Foreman::SettingManager.define(:test_context) do
+      category(:general) do
+        setting(:foo,
+          default: 'bar',
+          description: 'This is nicely described foo setting',
+          full_name: 'Foo setting')
+      end
+    end
+    assert_not_nil setting_memo['foo']
+    assert_equal setting_memo['foo'][:category], 'general'
+  end
+
+  it 'adds setting to defined category within block and sets it\'s label' do
+    Foreman::SettingManager.define(:test_context) do
+      category(:my_category, 'Awesome Category') do
+        setting(:foo,
+          default: 'bar',
+          description: 'This is nicely described foo setting',
+          full_name: 'Foo setting')
+      end
+    end
+    assert_not_nil setting_memo['foo']
+    assert_equal setting_memo['foo'][:category], 'my_category'
+    assert_equal category_memo['my_category'], 'Awesome Category'
+  end
+
+  it 'doesnt allow setting redefinition' do
+    assert_raise ::Foreman::Exception do
```

Comment:
Should the exception itself be verified as well? `Foreman::Exception` could be raised for a bunch of reasons.